### PR TITLE
group3

### PR DESCRIPTION
**Pull request by corebonts**

### CODE REVIEW

File: Java/src/main/java/com/gildedrose/QualityUpdater.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package com.gildedrose;
+
+abstract class QualityUpdater {
```
Comment (zopad) [inline Java/src/main/java/com/gildedrose/QualityUpdater.java:3]: Very cool and extendable.

File: Java/src/main/java/com/gildedrose/DefaultQualityUpdater.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package com.gildedrose;
+
+public class DefaultQualityUpdater extends QualityUpdater {
+    @Override
+    public boolean handlesItem(Item item) {
+        return true;
+    }
+    @Override
+    public void updateQuality(Item item){
+        decreaseQuality(item);
```
Comment (zopad) [inline Java/src/main/java/com/gildedrose/DefaultQualityUpdater.java:10]: This encapsulates the "business logic" of an item that is losing quality very well.

File: Java/src/main/java/com/gildedrose/GildedRose.java [hunk 3]
```diff
@@ -1,62 +1,25 @@
...
-                    if (items[i].quality < 50) {
-                        items[i].quality = items[i].quality + 1;
-                    }
-                }
-            }
+            Item item = items[i];
+            qualityUpdaters.stream()
+                .filter(qualityUpdater -> qualityUpdater.handlesItem(item)).findFirst()
+                .orElse(defaultQualityUpdater)
+                .updateQuality(item);;
```
Comment (zopad) [inline Java/src/main/java/com/gildedrose/GildedRose.java:22]: Insanely minor: double semicolon :P

Comment (TehMillhouse) [inline Java/src/main/java/com/gildedrose/GildedRose.java:22]: 😘 👌

**Review by TehMillhouse — COMMENTED** [review]
I guess all I have to say is I'm jealous.

**Pull request merged**